src/client/components/search-component/index.jsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import './index.scss';
+import { Input, Button, Form, Dropdown, Header } from 'semantic-ui-react';
+import { DatesRangeInput } from 'semantic-ui-calendar-react';
+import moment from 'moment';
+
+class SearchComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            destination: '',
+            date: '',
+            rooms: 1,
+            adults: 2,
+            children: 0
+        };
+    }
+
+    handleChange(event, {name, value}) {
+        this.setState({ [name]: value });
+    }
+
+    toggleRoomSelector(){
+        document.getElementById('room-selector').classList.toggle('hidden');
```

Comment:
It's a very rare situation when you need to use ids in React. Instead, you should always use refs. Read more https://reactjs.org/docs/refs-and-the-dom.html
Please replace the logic based on ids with the one based on refs everywhere in the component.